### PR TITLE
[shopsys] role `ROLE_ALL_API` sees all customer user orders from common customer

### PR DESCRIPTION
**Pull request by malyMiso**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Customer user with role  `ROLE_ALL_API` can see all orders created by customer users of same customer.  
|New feature| Yes 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

----
:globe_with_meridians: Live Preview:
  - https://mt-customer-orders.odin.shopsys.cloud
  - https://cz.mt-customer-orders.odin.shopsys.cloud

### CODE REVIEW

File: packages/frontend-api/src/Model/Order/OrderApiFacade.php [hunk 1]
```diff
@@ -146,4 +146,24 @@ public function getCustomerOrderCount(Customer $customer): int
...
+    {
+        return $this->orderRepository->getByUuidAndCustomer($uuid, $customer);
+    }
+
+    /**
+     * @param string $orderNumber
+     * @param \Shopsys\FrameworkBundle\Model\Customer\Customer $getCustomer
+     * @return \Shopsys\FrameworkBundle\Model\Order\Order
+     */
+    public function getByOrderNumberAndCustomer(string $orderNumber, Customer $getCustomer): Order
```
Comment (grossmannmartin) [inline packages/frontend-api/src/Model/Order/OrderApiFacade.php]: argument should be named just `$customer`, or?

File: upgrade-notes/backend_20240731_200701.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#### [shopsys] role `ROLE_ALL_API` sees all customer user orders from common customer ([#3296](https://github.com/shopsys/shopsys/pull/3296))
```
Comment (grossmannmartin) [inline upgrade-notes/backend_20240731_200701.md]: please without `[shopsys]` and preferably in a form like "do this" - e.g., "allow ROLE_API_ALL to see all customer user orders from common customer", or something like that

**Review by grossmannmartin — APPROVED** [review]

File: packages/frontend-api/src/Model/Order/OrderApiFacade.php (hunk 1 above)
Comment (malyMiso) [reply packages/frontend-api/src/Model/Order/OrderApiFacade.php]: fixed: https://github.com/shopsys/shopsys/pull/3296/commits/3483d444204f23c001c587016a6886bef7030342

File: upgrade-notes/backend_20240731_200701.md (hunk 2 above)
Comment (malyMiso) [reply upgrade-notes/backend_20240731_200701.md]: Thx. Fixed: https://github.com/shopsys/shopsys/pull/3296/commits/98de48f94c15fa838c9ed80ff49ab5ee47330cbd

**Review by pk16011990 — APPROVED** [review]

**Pull request merged**